### ISSUE TITLE
chef-client runs with version 18.x get a EOL warning

### ISSUE THREAD

**Issue by Ladybiss** [chef/chef#14365]
## Description
on 18.4.12, the last published version, chef-client run gives a EOL warning:
```
[2024-05-01T11:47:05+00:00] WARN: This release of Chef Infra Client became end of life (EOL) on May 1st 2024. Please update to a supported release to receive new features, bug fixes, and security updates.
```

## Chef Version

18.4.12 or 18.4.2
## Platform Version

Ubuntu 22.04
## Replication Case

See: https://github.com/chef/chef/blob/main/lib/chef/client.rb#L339

```
    def warn_if_eol
      require_relative "version"

      # We make a release every year so take the version you're on + 2006 and you get
      # the year it goes EOL
      eol_year = 2006 + Gem::Version.new(Chef::VERSION).segments.first

      if Time.now > Time.new(eol_year, 5, 01)
        logger.warn("This release of #{ChefUtils::Dist::Infra::PRODUCT} became end of life (EOL) on May 1st #{eol_year}. Please update to a supported release to receive new features, bug fixes, and security updates.")
      end
    end
```

## Client Output

```
[2024-05-01T11:47:05+00:00] WARN: This release of Chef Infra Client became end of life (EOL) on May 1st 2024. Please update to a supported release to receive new features, bug fixes, and security updates.
```

**Comment by YorithTheDreamer** [comment]
Can confirm, this is happening on every run on my end as well.

**Comment by the6thBook** [comment]
RHEL9 as well

**Comment by nareshsukhija** [comment]
This happens on every run for all current RHEL versions (7/8/9), kindly treat this as an urgent issue.

**Comment by breisig** [comment]
This is a HUGE issue on RHEL 9...

**Comment by blackknight36** [comment]
Seeing the same message on Ubuntu 20.04.

**Comment by breisig** [comment]
This should be simple remove and a quick release needs to happen.

**Comment by tpowell-progress** [comment]
This should be in our next 18 release. Release date still pending.

**Comment by randyrue** [comment]
Any updates on an 18.5 release date?

**Comment by fern4lvarez** [comment]
I think this can be closed, the fix is included in 18.5.0: https://discourse.chef.io/t/chef-infra-client-18-5-0-released/23188

**Comment by Ladybiss** [comment]
Confirmed it's working as expected, no more warnings once chef-client is updated to 18.5
